munit-cats-effect/shared/src/main/scala/munit/CatsEffectFunFixtures.scala

```diff
@@ -0,0 +1,55 @@
+package munit
+
+import cats.effect.{IO, Resource}
+import cats.syntax.flatMap._
+
+import scala.concurrent.Promise
+
+trait CatsEffectFunFixtures extends FunFixtures { self: CatsEffectSuite =>
+
+  object CatsEffectFixture {
+
+    def fromResource[T](
+        resource: Resource[IO, T]
+    ): FunFixture[T] = fromResource(
+      resource,
+      (_, _) => IO.pure(()),
+      _ => IO.pure(())
+    )
+
+    def fromResource[T](
+        resource: Resource[IO, T],
+        setup: (TestOptions, T) => IO[Unit],
+        teardown: T => IO[Unit]
+    ): FunFixture[T] = {
+      val promise = Promise[IO[Unit]]()
+
+      FunFixture.async(
+        setup = { testOptions =>
+          implicit val ec = munitExecutionContext
```

Comment:
Is it OK/expected that `munitExecutionContext` is parasitic by default? Users can always override it if they want something else. I felt that a parasitic EC was the best default to 1) have deterministic test behaviors and 2) keep stack traces clean.

Reply:
I don't see any issue with the parasitic `EC` :)

Reply:
So, as it turns out, this causes deadlock so you are forced to override it. It also doesn't make for a great default imho, because the failure mode is "oh no there is a deadlock", and you don't expect it to come from your test framework and it can be baffling

Reply:
@systemfw Could you please open an issue here? https://github.com/typelevel/munit-cats-effect/

Reply:
that's where I'm coming from https://github.com/typelevel/munit-cats-effect/pull/65

I don't expect munit to take any action at this point (I would have opened a PR), but I thought it was interesting to share if someone else comes across the discussion :)

Reply:
OK, disregard my comment then :)